src/algorithms/GenericFunctions.jl

```diff
@@ -0,0 +1,31 @@
+###############################################################################
+#
+#   GenericFunctions.jl : Functions for Generic types
+#
+###############################################################################
+
+# TODO: Move from CRT from Hecke to AbstractAlgebra?
+# Currently no implementation, only example on how the arbitrary inputs `crt`
+# should look like.
+@doc Markdown.doc"""
+    crt(r::Array{T, 1}, m::Array{T, 1}) where T
+    crt(r::T, m::T...) where T
+
+Return $x$ in the Euclidean domain $T$ such that $x \equiv r_i \mod m_i$
+for all $i$.
+"""
+function crt end
+
+@doc Markdown.doc"""
+    factor(a::T, b::R)
```

Comment:
Is `b` supposed to be a ring containing `a`? Then the documentation should probably say so.

Reply:
As written below, I don't know anything of these things. Looked at `Hecke` to write something.